py/desispec/test/integration_test_quicklook.py

```diff
@@ -0,0 +1,176 @@
+            if runcmd(cmd, clobber=clobber) != 0:
+                raise RuntimeError('desi_extract_spectra failed for camera r0')
+
+            cmd = "desi_extract_spectra -i {}/pix-z0-00000001.fits -o {}/frame-z0-00000001.fits -f {}/fibermap-00000001.fits -p {} -w 7650,9830,0.8 -n {}".format(os.path.dirname(simspec),os.path.dirname(simspec),os.path.dirname(simspec),psf_z,nspec)
+            if runcmd(cmd, clobber=clobber) != 0:
+                raise RuntimeError('desi_extract_spectra failed for camera z0')
+
+    for camera in ['b0','r0','z0']:
+
+        cmd = "desi_compute_fiberflat --infile {}/frame-{}-00000001.fits --outfile {}/fiberflat-{}-00000001.fits".format(os.path.dirname(simspec),camera,os.path.dirname(simspec),camera)
+        if runcmd(cmd, clobber=clobber) != 0:
+            raise RuntimeError('desi_compute_fiberflat failed for camera {}'.format(camera))
+
+        cmd = "desi_bootcalib --fiberflat {}/pix-{}-00000001.fits --arcfile {}/pix-{}-00000000.fits --outfile {}/psfboot-{}.fits".format(os.path.dirname(simspec),camera,os.path.dirname(simspec),camera,os.path.dirname(simspec),camera)
+        if runcmd(cmd, clobber=clobber) != 0:
+            raise RuntimeError('desi_bootcalib failed for camera {}'.format(camera))
+
+    return
+
+def integration_test(night="20160728", nspec=5, clobber=False):
+    """Run an integration test from raw data simulations through quicklook pipeline
+
+    Args:
+        night (str): YEARMMDD
+        nspec (int): number of spectra to include
+        clobber (bool, optional): rerun steps even if outputs already exist
+
+    Raises:
+        RuntimeError if quicklook fails
+    """
+    log = logging.get_logger()
+    log.setLevel(logging.DEBUG)
+    datadir = os.path.join('$DESI_SPECTRO_SIM','exposures')
+
+    expid = 2
+    flat_expid = 1
+
+    # check for required environment variables and simulate inputs
+    check_env()
+    sim(night=night, nspec=nspec, clobber=False)
+
+    for camera in ['r0','z0']:
+
+        # find all necessary input and output files
+        psffile = os.path.join(datadir,night,'psfboot-{}.fits'.format(camera))
+        fiberflatfile = os.path.join(datadir,night,'fiberflat-{}-{:08d}.fits'.format(camera,flat_expid))
+
+        # verify that quicklook pipeline runs
+        com = "desi_quicklook -n {} -c {} -e {} -f {} --psfboot {} --fiberflat {} --rawdata_dir {} --specprod_dir {}".format(night,camera,expid,'dark',psffile,fiberflatfile,datadir,'$DESI_SPECTRO_SIM')
+        if runcmd(com, clobber=clobber) != 0:
+            raise RuntimeError('quicklook pipeline failed for camera {}'.format(camera))
+
+        os.remove('lastframe-{}-{:08d}.fits'.format(camera,expid))
+
+    # remove all output
+    output_dir = os.path.join(os.environ['DESI_SPECTRO_SIM'],'exposures',night)
```

Comment:
Please add an option for whether the output files should be cleaned up or not.  It can be useful to run an integration test and have the output say behind to inspect.

Normally raw data simulations should be in `$DESI_SPECTRO_SIM/$PIXPROD`, with any processing of those simulations in a separate directory such as `$DESI_QUICKLOOK_DIR` or `$DESI_SPECTRO_REDUX/$QL_DIR`.  Quicklook should not write output into the raw data directory, and it shouldn't know that your simulated raw data aren't actually real raw data.

This cleanup as-is could blow away files that existed prior to the test being run.   Please isolate the test output (simulation and processing) into separate directories and raise an error if they exist at the start of the test (or provide a clobber option that must be set to overwrite a pre-existing directory).  Then you can safely cleanup those directories.

Note: the normal DESI spectro pipeline integration test also doesn't isolate its output into a test directory (bad) but at least it doesn't auto-cleanup everything and risk blowing something else away (good).

Cleanup should also be structured such that it will run (if requested) even if earlier code raises an exception.  e.g.
```python
try:
    # run the tests...
except Exception as e:
    print('Gack! Something failed while running the tests')
    raise e
finally:
    #- always excecuted, even upon exception
    if cleanup:
        # cleanup files
```